### PR TITLE
Enhance Google Analytics

### LINKED ISSUE #89: Enhance Google Analytics

### PR DESCRIPTION
**Pull request by alisonmonteiro**
It aims to fix #89 

Kap is tracking `Recording ended` already. I used this as an example 😉 

#### Tracked events

- [x]   Opened Preferences
- [x]   Toggled Settings
- [x]   Opened Plugin Configuration
- [x]   Recording started
- [x]   Recording ended
- [x]   Accessed Editor/Preview
- [x]   Saved file
- [x]   Discarded file
- [x]   Exported
- [x]   Checked for updates

### CODE REVIEW

File: app/src/common/settings-manager.js [hunk 1]
```diff
@@ -76,6 +77,10 @@ export const set = (key, value, {volatile = false} = {}) => {
     return objectPath.set(volatiles, key, value);
   }
 
+  if (key !== 'dimensions') {
+    track(`settings/${key}/toggled`);
```
Comment (sindresorhus) [inline app/src/common/settings-manager.js]: Toggled is only useful if we know whether it was enabled or disabled.

File: app/src/main/index.js [hunk 2]
```diff
@@ -312,10 +314,14 @@ const openPrefsWindow = () => {
 
   prefsWindow.on('close', () => {
     prefsWindow = undefined;
+    track('preferences/closed');
```
Comment (sindresorhus) [inline app/src/main/index.js]: I don't see how this is useful. We already know it was opened. Of course the user is going to close it too. Unless I'm missing something.

File: app/src/main/menus.js [hunk 3]
```diff
@@ -11,6 +12,8 @@ const checkForUpdatesItem = {
         title: 'No updates available!',
         body: 'You will automatically receive updates as soon as they are available 🤗'
       })).show();
+
+      track('updates/checked');
```
Comment (sindresorhus) [inline app/src/main/menus.js]: This should rather be more specific for reporting when users click the "Check for Updates" button. Reporting when an update is checked for is not very useful.

**Comment by sindresorhus** [comment]
Some things that could be useful to know:

- What plugins users install so we can see which one is the most popular.
- What plugins users use in the export menu.
- What FPS users set.
- What file formats users use.

**Comment by sindresorhus** [comment]
> Recording started
  Recording ended

What we want to know is how many recordings users are doing and how long. Just events for start and stop are not that useful.

**Comment by skllcrn** [comment]
This is really helpful and a great first contribution, thank you @alisonmonteiro!

**Comment by alisonmonteiro** [comment]
@skllcrn @sindresorhus thanks for the feedback! I'll send an update soon. 🙌

**Comment by alisonmonteiro** [comment]
> What we want to know is how many recordings users are doing and how long.

I think that [`15835ee`](https://github.com/wulkano/kap/pull/463/commits/15835eea065842c5439559a3de477d5894947f98) solves this.

> What plugins users install so we can see which one is the most popular.
What plugins users use in the export menu.
What FPS users set.
What file formats users use.

I can do it. It will take more time for me. But I can do. 😅

File: app/src/main/index.js [hunk 4]
```diff
@@ -513,6 +513,9 @@ ipcMain.on('will-start-recording', () => {
     cropperWindow.setAlwaysOnTop(true);
   }
 
+  timeStartedRecording = new Date().getTime();
```
Comment (sindresorhus) [inline app/src/main/index.js]: `Date.now()` ;)

**Comment by skllcrn** [comment]
Would it be possible to know which version of Kap the events are coming from, preferably without losing interoperability with previous events?

**Comment by alisonmonteiro** [comment]
> Would it be possible to know which version of Kap the events are coming from

By adding a prefix?

`v2.1.2/preferences/opened`
`v2.1.2/file/discarded`

or simply

`v2/preferences/opened`
`v2/file/discarded`

🤔

**Comment by skllcrn** [comment]
Definitively prefer the incremental semver if that's possible!

**Comment by sindresorhus** [comment]
@skllcrn I don't think we should differentiate patch versions though. Then we get too many disparate events. We could do `v2.1`, `v2.2`, etc.

**Comment by alisonmonteiro** [comment]
> We could do v2.1, v2.2, etc.

Yes, cool

We could use [`semver-truncate`](https://github.com/sindresorhus/semver-truncate) and

```js
const truncate = semverTruncate('1.3.5-foo', 'minor');
//=> '1.3.0'
const version = `v${truncate.slice(0, -2)}`
//=> 'v1.3'
```

And I think we don't even need the `.slice` here.

🤔

**Comment by alisonmonteiro** [comment]
If there is anything more I can do, so we can get this merged, please let me know. 🙌

File: app/src/main/analytics.js [hunk 5]
```diff
@@ -1,18 +1,20 @@
 import firstRun from 'first-run';
 import Insight from 'insight';
+import {parse} from 'semver';
 import pkg from '../../package';
 import {get as getSetting} from '../common/settings-manager';
 
 const trackingCode = 'UA-84705099-2';
 const insight = new Insight({trackingCode, pkg});
+const version = parse(pkg.version, {loose: true});
```
Comment (sindresorhus) [inline app/src/main/analytics.js]: No need to use the `loose` option. We follow semver.

Comment (alisonmonteiro) [reply app/src/main/analytics.js]: Oh, I see. Thanks again for the feedback. 🙌

**Comment by sindresorhus** [comment]
Excellent. Thank you, @alisonmonteiro :)

**Pull request merged**